Another batch of IDEA-prompted changes

voidsong-dragonfly opened the pull request:
Lots of unused variables went away. If possible, could you look this over? I already did but a second pair of eyes always helps.

voidsong-dragonfly committed:
Another batch of IDEA-prompted changes

voidsong-dragonfly committed:
Multiple bugfixes

Brycey92 commented on src/main/java/zmaster587/advancedRocketry/client/render/planet/RenderPlanetarySky.java:
```diff
@@ -207,22 +207,16 @@ public void render(float partialTicks, WorldClient world, Minecraft mc) {
 			if (primaryStar != null)
 			{
 				sunSize = properties.getStar().getSize();
-				starSeparation = properties.getStar().getStarSeparation();
```
Do these variables no longer factor into rendering the sky? And should they?

voidsong-dragonfly replied:
Those never had any use at all.

voidsong-dragonfly replied:
As in, IDEA specifically said they were not used, and nothing caused problems when I straight-up removed em. So IDEA wasn't lying in that case.

Brycey92 commented on src/main/java/zmaster587/advancedRocketry/client/render/planet/RenderSpaceTravelSky.java:
```diff
@@ -569,14 +563,12 @@ private void buildSolarSystem(SpacePosition playerPosition)
 				float phaseInc = 360f / subStars.size();
 				double phase = 0;
 				for(StellarBody subStar : subStars) {
-					float solarOrbitalDistance = 40 * subStar.getStarSeparation();
 
 					//Radius to put the player
-					double radius = solarOrbitalDistance;
```
Perhaps sum up these 3 lines of variable names and comments into a new comment on the calculation so it's less magical math next time someone needs to look at it. Also, don't forget to delete the dangling comment here! :P

voidsong-dragonfly replied:
Coming in next commit

Brycey92 commented on src/main/java/zmaster587/advancedRocketry/dimension/DimensionManager.java:
```diff
@@ -1090,14 +1089,11 @@ public void createAndLoadDimensions(boolean resetFromXml)
 			//Silence you fool!
 			//Patch to fix JEI printing when trying to load planets too early
 			return loadedDimProps;
-		} catch (FileNotFoundException e) {
```
It looks like you meant to merge the two catch branches. If that's the case, why was this exception not added to the other branch?

voidsong-dragonfly replied:
FileNotFound is a subclass of the other one

Brycey92 commented on src/main/java/zmaster587/advancedRocketry/entity/EntityHoverCraft.java:
```diff
@@ -85,10 +85,7 @@ public void markDirty() {
 
 	public void updateRiderPosition()
 	{
-		if (!this.getPassengers().isEmpty())
```
Does getPassengers() have any side effects from being called? (Ignoring what IDEA says - it thinks a lot of methods have side effects when they don't really.)

Also, should this method be doing something, or can it be deleted entirely?

voidsong-dragonfly replied:
It's literally just a list get. It does not.

Brycey92 replied:
Removal coming in next commit?

voidsong-dragonfly replied:
Yeah

Brycey92 commented on src/main/java/zmaster587/advancedRocketry/entity/fx/RocketFx.java:
```diff
@@ -48,20 +48,12 @@ public int getFXLayer() {
-        float f1 = f + 0.0624375F;
-        float f2 = (float)this.particleTextureIndexY / 16.0F;
-        float f3 = f2 + 0.0624375F;
+        float f;
+        float f1;
+        float f2;
+        float f3;
         float f4 = 0.1F * this.particleScale;
 
-        if (this.particleTexture != null)
-        {
-            f = this.particleTexture.getMinU();
-            f1 = this.particleTexture.getMaxU();
-            f2 = this.particleTexture.getMinV();
-            f3 = this.particleTexture.getMaxV();
-        }
```
I wonder if either of these produce better effects than the assignment of 0s and 1s to the variables later on... When I tried launching a rocket, the smoke effects seemed screwy to me. Maybe these could help that?

voidsong-dragonfly replied:
They've always been sort of strange, it's something I need to look into

Brycey92 commented on src/main/java/zmaster587/advancedRocketry/inventory/GuiHandler.java:
```diff
@@ -57,22 +50,14 @@ else if(x == -1) {
-			tile = player.getHeldItem(EnumHand.MAIN_HAND).getItem();
 		}
-		else
-			tile = world.getEntityByID(x);
```
What on God's green Earth was the original developer trying to do with tile throughout the whole function? I am both intrigued and baffled.

voidsong-dragonfly replied:
Dunno, ask Z

Brycey92 commented on src/main/java/zmaster587/advancedRocketry/item/ItemOreScanner.java:
```diff
@@ -121,9 +121,8 @@ public void interactSatellite(SatelliteBase satellite,EntityPlayer player, World
 
 	@Override
 	public List<ModuleBase> getModules(int id, EntityPlayer player) {
-		List<ModuleBase> modules = new LinkedList<>();
 		//modules.add(new ModuleOreMapper(0, 0));
-		return modules;
```
I had originally left this alone to maintain the context of the comment. If we want to change it, we should find out what that piece of code would have done for us and either implement it or delete it.

voidsong-dragonfly replied:
Oh right that was my own comment - nothing, but it was halfway to using modular for this. I'll leave it in

voidsong-dragonfly replied:
Coming in next commit

voidsong-dragonfly committed:
Fix warp core

voidsong-dragonfly committed:
Fix planet gen NPE

voidsong-dragonfly committed:
Fix data bus issue

voidsong-dragonfly committed:
More fixes!

 - Remove more problematic alt recipes
 - More cleanup
 - Remove useless/already resolved todos
 - Remove some more useless files
 - Remove multiple calls to deprecated methods and replace with correct one
 - Dedeprecate StorageChunk#getGuidanceComputer because there's no alternative and that works, plus the commit depping it doesn't make sense to do so

Brycey92 commented on src/main/java/zmaster587/advancedRocketry/satellite/SatelliteBiomeChanger.java:
```diff
@@ -89,7 +89,7 @@ public void tickEntity() {
 		if(world != null) {
 
 			for(int i = 0; i < 10; i++) {
-				if(world.getTotalWorldTime() % 1 == 0 && !toChangeList.isEmpty()) {
```
This looks like a failed attempt to limit the speed of operation of the Biome Changer. If its current speed is performant and has a power draw you're happy with, this can be safely removed.

voidsong-dragonfly replied:
I think it's limited by power draw anyway, which is a better way to do it. I'd need to check that, though

voidsong-dragonfly replied:
Just fixed a bug and left it. Want to redo later

Brycey92 commented on src/main/java/zmaster587/advancedRocketry/tile/TileForceFieldProjector.java:
```diff
@@ -34,7 +33,8 @@ public void update() {
 		if(world.getTotalWorldTime() % 5 == 0) {
 			if(world.isBlockPowered(getPos())) {
-				if(extensionRange < MAX_RANGE) {
+                short MAX_RANGE = 32;
```
Same thing here about using `final` if this is never assigned to again.

voidsong-dragonfly replied:
Tbh what IDEA _should_ have done is made these final and not moved them down... can you search for any more like that I'd want to fix?

Brycey92 commented on src/main/java/zmaster587/advancedRocketry/tile/TilePump.java:
```diff
@@ -162,7 +161,8 @@ private BlockPos getNextBlockLocation()
 		{
 			BlockPos nextElement = queue.poll();
-			if(visited.contains(nextElement) || nextElement.getDistance(pos.getX(), nextElement.getY(), pos.getZ()) > RANGE )
+			int RANGE = 64;
```
`final` :P

voidsong-dragonfly replied:
Same as above

Brycey92 commented on src/main/java/zmaster587/advancedRocketry/tile/TileRocketAssemblingMachine.java:
```diff
@@ -190,6 +185,7 @@ public boolean shouldRenderInPass(int pass) {
 
 	@Override
 	public int getPowerPerOperation() {
+		int ENERGYFOROP = 100;
```
This can probably just be inlined

voidsong-dragonfly replied:
I decided to just make it a final variable at the top because it was there originally and it made sense

Brycey92 commented on src/main/java/zmaster587/advancedRocketry/tile/TileRocketAssemblingMachine.java:
```diff
@@ -501,6 +496,7 @@ public AxisAlignedBB getRocketPadBounds(World world, BlockPos pos) {
 			return null;
 
 		//Get min and maximum Z/X bounds
+		int MAX_SIZE = 16;
```
`final`?

voidsong-dragonfly replied:
Same as other "final" ones

Brycey92 commented on src/main/java/zmaster587/advancedRocketry/tile/TileRocketAssemblingMachine.java:
```diff
@@ -539,6 +535,7 @@ public AxisAlignedBB getRocketPadBounds(World world, BlockPos pos) {
 
 		int maxTowerSize = 0;
 		//Check perimeter for structureBlocks and get the size
+		int MAX_SIZE_Y = 64;
```
`final`!

voidsong-dragonfly replied:
Yeah, IDEA was being annoying. I thought it would be smarter :(

Brycey92 commented on src/main/java/zmaster587/advancedRocketry/tile/TileRocketAssemblingMachine.java:
```diff
@@ -560,6 +557,8 @@ public AxisAlignedBB getRocketPadBounds(World world, BlockPos pos) {
 		}
 
 		//if tower does not meet criteria then reutrn null
+		int MIN_SIZE_Y = 4;
+		int MIN_SIZE = 3;
```
`final`, `final` xD

voidsong-dragonfly replied:
IDEAAAA :(

Brycey92 commented on src/main/java/zmaster587/advancedRocketry/tile/infrastructure/TileRocketFluidLoader.java:
```diff
@@ -93,6 +91,7 @@ public boolean allowRedstoneOutputOnSide(EnumFacing facing) {
 
 	protected boolean getStrongPowerForSides(World world, BlockPos pos) {
 		for(int i = 0; i < 6; i++) {
+			int ALLOW_REDSTONEOUT = 2;
```
`final` here too please!

voidsong-dragonfly replied:
IDEAAAA :(

Brycey92 commented on src/main/java/zmaster587/advancedRocketry/tile/infrastructure/TileRocketLoader.java:
```diff
@@ -104,6 +102,7 @@ public boolean allowRedstoneOutputOnSide(EnumFacing facing) {
 
 	protected boolean getStrongPowerForSides(World world, BlockPos pos) {
 		for(int i = 0; i < 6; i++) {
+			int ALLOW_REDSTONEOUT = 2;
```
ditto

voidsong-dragonfly replied:
IDEAAAA :(

Brycey92 commented on src/main/java/zmaster587/advancedRocketry/world/ChunkProviderAsteroids.java:
```diff
@@ -31,8 +31,7 @@
 	private final World world;
-	private final boolean generateStructures;
-	private final Random rand;
+    private final Random rand;
```
Why are all your changes in this file appearing shifted to the left by 2 spaces? Don't tell me this file uses a different whitespace type from every other file in the project...
 .·´¯(>▂<)´¯·.

voidsong-dragonfly replied:
Dunno

Brycey92 commented on src/main/java/zmaster587/advancedRocketry/dimension/DimensionProperties.java:
```diff
@@ -1734,7 +1727,7 @@ public void writeToNBT(NBTTagCompound nbt) {
 		nbt.setBoolean("isGasGiant", isGasGiant);
 		nbt.setBoolean("hasRings", hasRings);
-		nbt.setInteger("seaLevel", sealevel);
+		nbt.setInteger("sealevel", sealevel);
```
Maybe I could have you rename the variable to seaLevel, if not the NBT tag?

voidsong-dragonfly replied:
Sure!

Brycey92 commented on src/main/java/zmaster587/advancedRocketry/world/ChunkManagerPlanet.java:
```diff
@@ -197,31 +197,27 @@ public void resetCache() {
 
 	@Nonnull
-	public Biome[] getBiomesForGeneration(@Nonnull Biome[] biomes, int x, int z, int width, int height)
+	public Biome[] getBiomesForGeneration(Biome[] biomes, int x, int z, int width, int height)
```
If this takes null, let's stick `@Nullable` in front of the first parameter

Brycey92 commented on src/main/java/zmaster587/advancedRocketry/item/ItemStationChip.java:
```diff
@@ -272,11 +271,7 @@ public void setSelectionId(@Nonnull ItemStack stack, int dimid, int slotId)
-					try {
-						retList.add(LandingLocation.loadFromNBT((NBTTagCompound)tag));
-					} catch (NoSuchAttributeException e) {
-						AdvancedRocketry.logger.warn("Attempting to load a Landing location for planet " + dimid + " but chip appears to have malformed data");
```
Perhaps we ought to make sure we put this check where it's possible to get broken NBT from a save file

voidsong-dragonfly replied:
Readded it, you're right

voidsong-dragonfly committed:
First batch of fixes

voidsong-dragonfly committed:
Second batch of fixes, with many IDEA mixups fixed

voidsong-dragonfly committed:
Merge branch '1.12-cleanup' into bryce_cleanup
